gapir/cc/memory_allocator.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GAPIR_MEMORY_ALLOCATOR_H
```

Comment:
I would suggest adding here (top of header) a short paragraph of comments for an overview covering:

why: cannot rely on system allocator
how: single big malloc, then manage two kinds of memories (static and purgeable), LRU based eviction, compaction.

Reply:
Done